backup/queries_statistics.go

```diff
@@ -119,13 +120,20 @@ func GetAttributeStatistics(connectionPool *dbconn.DBConn, tables []Table) map[u
 		inheritClause, statSlotClause, statCollationClause,
 		SchemaFilterClause("n"), utils.SliceToQuotedString(tablenames))
 
-	results := make([]AttributeStatistic, 0)
-	err := connectionPool.Select(&results, query)
+	return connectionPool.Query(query)
+}
+
+func GetAttributeStatistics(connectionPool *dbconn.DBConn, tables []Table) map[uint32][]AttributeStatistic {
```

Comment:
For `GetAttributeStatistics` and `GetTupleStatistics` I would suggest to add comments describing why they should not be used in the main gpbackup flow (to prevent somebody using them in the future), and the purpose why they are not removed completely.

Reply:
rewritten via callbacks